### DISCUSSION TITLE
Graphql+Prisma vs Prisma

### DISCUSSION THREAD

**Discussion by aramis-it** [prisma/prisma#4843] [Q&A]
What are advantages  using GraphQL with Prisma instead just Prisma. Why we need GraphQL?

For example. // Following queries not same
Its usage almost same for me. Why we need graphQL?
```
Graphql
query GetSomeUsers {
  users(where: { email: { contains: "prisma" } }, orderBy: { name: desc }) {
    id
    name
    email
    posts(take: 10, orderBy: { updatedAt: desc }) {
      published
      title
      content
    }
  }
}
```
```
Prisma
{
    where: { id: 2 },
    data: { published: true },
}
```

**Comment by HendrikJan** [comment]
Hi Ulzi,

I try to answer you question, but it will not be complete.
Hopefully it will help you enough to start understanding the difference between Prisma and Graphql.

Prisma and Graphql are used in different situations.

**Situation: If you only work on the server and the user of the application is allowed to access all data from the database**
If you only work inside NodeJs, then the only thing you need is Prisma. For instance an application that reads from your database, does some calculations and then displays the result in the terminal on the command line.

In this case you create a single application.

**Situation: If you want to restrict access to some data**
In case the user is not allowed to view all data you can restrict access to some data in your database using Graphql.
For instance an application where the user logs in and may only see her/his own information and not that of other persons.

In this case you need to build two applications.
1. The server. The server will get a Graphql query from the client, read data from the database using Prisma, transform it to Graphql, and then send the result to the client.
2. The client is where the user logs in. The client will make a Graphql query, sent it to the server, receive the result and display it to the user.

**Situation: When the application is a website**
If your application is a website that needs information that is on the database on the server.
A website cannot use Prisma (to my knowledge) to get information from the server. A website needs to send a request to the server an then the server sends a response back with the requested data.

Graphql is a way to build queries and responses (in which case you use Prisma+Graphql).
You can choose another way, for example REST (in which case you use Prisma+REST).

So in the case of a website you also build two applications:
1. The server on NodeJs
2. The client in your browser

**So what is it that Graphql adds?**
To say it another way:

Prisma is *only* used to talk to the database.

Graphq adds features like:
- security
- restricting access
- transform the data
- combining data from multiple databases and other sources
- send data between a server and a browser
- allow a team of developers to split up in a server-team and a website-team
- create a server with Graphql for which other companies can build a website

**Reply by aramis-it** [reply]
Thank you very much.
This part clear for me.

```
Graphq adds features like:

- security
- restricting access
- transform the data
- combining data from multiple databases and other sources
- send data between a server and a browser
- allow a team of developers to split up in a server-team and a website-team
- create a server with Graphql for which other companies can build a website
```

**Reply by HendrikJan** [reply]
:-)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
